lib/index.js

```diff
@@ -199,6 +200,17 @@ exports.Socket = function (type) {
     }
   }
 
+  if (this.type == 'pull') {
+    this._pause = false;
```

Comment:
In my opinion performance should be slightly different, but the point that convinced me was:
> the thing here is that you could check any socket type with console.log and it will have a _pause property set to false.

So for a different socket type than `pull` this property could be misused.